src/server/tiering/op_manager.h

```diff
@@ -34,6 +34,9 @@ class OpManager {
   // Schedule read for offloaded entry that will resolve the future
   util::fb2::Future<std::string> Read(EntryId id, DiskSegment segment);
 
+  // Modify offloaded entry, does not change stored value, but affects succeeding reads
```

Comment:
why did you choose such API?  
why not extend Read with an optional post-processing callback?

Reply:
It's simpler to reason about if they are separate. Reads keep simple, modifies do as well

Reply:
From the API perspective, you introduce a state between calls. 
does it self clear after the subsequent read? is it a constant thing?

Reply:
Sorry, "no" - as it does not self-clear?
So if you pass a callback capturing some variable from stack/request, it will be called on all subsequent Reads?

Reply:
No, it will be called once and visible to all succeeding reads or operations

Reply:
that's like a linear queue of reads or writes